### PR TITLE
Migration of l10n_ch_bank and l10n_ch_zip

### PR DESCRIPTION
**Pull request by guewen**
Supersedes https://github.com/OCA/l10n-switzerland/pull/62

I worked on top of the @BT-olivier's branch and changed those things:
- Cleanings (usage of string literals instead of docstrings, ...)
- Use the new API
- The wizard to update the banks do no longer needs to create a temporary file
- The xmlids for the banks must be created in the l10n_ch_bank namespace, not in l10n_ch,
  correct the history with a migration
- Convert date fields to fields.Date
- Add again the record which where in the 7.0 files but add them as 'active=False' so
  they for the ones who migrate, they will be deactivated instead of deleted
- remove the *_old modules
- restore the `ir.country.state` which have been dropped during the module's migration
- The xmlids of the `res.better.zip` have been changed from the 7.0 version. I restored all of them (using this script https://gist.github.com/guewen/34fbc497297a1af0ca2f)

### CODE REVIEW

File: l10n_ch_bank/__openerp__.py [hunk 1]
```diff
@@ -18,23 +19,22 @@
...
-but can improve the user experience.
-""",
- 'website': 'http://www.camptocamp.com',
- 'data': ['l10n_ch_better_zip.xml'],
- 'demo_xml': [],
- 'installable': False,
- 'active': False}
+{
+    'name': 'Switzerland - Bank list',
+    'version': '8.0.9.0.0',
```
Comment (yvaucher) [inline l10n_ch_bank/__openerp__.py:24]: Version number should be x.y.z

Comment (guewen) [inline l10n_ch_bank/__openerp__.py:24]: @yvaucher Nop. If you looks how odoo does internally, you will see that it prepends the release number with its own one (check the `latest_version`  field in `ir_module_module`), so if the version of a module is `1.2.3`, odoo will see it as `8.0.1.2.3`. Now, if the `8.0` part is added explicitly in `__openerp__.py` as I did, it understands it and won't add the `8.0` part.
The issue here is that the module version _was_ `8.0`, so odoo confused it with its own version and I couldn't get it to load my migration script, that's why I had to explicit the full version number.

Comment (open-net-sarl) [inline l10n_ch_bank/__openerp__.py:24]: Bonjour, 

Il semble que le commentaire de ce module soit celui de l10n_ch_zip, pas l10n_ch_bank !?!? 

----- Mail original -----

| In l10n_ch_bank/**openerp**.py: > -============================
| > -
| > - This module will load all Swiss postal codes (ZIP) in Odoo to >
| > -ease the input of partners.

Avec mes plus sincères salutations, 

Jean-Marc Vandel 
Senior Project Manager 

Open Net Sàrl 
Rue de l'Industrie 59 
CH-1030 Bussigny 

www.open-net.ch 
+41 79 212.65.94 
jmv@open-net.ch

Comment (guewen) [inline l10n_ch_bank/__openerp__.py:24]: @open-net-sarl No, it isn't. You have been confused by the git renaming which thinks that this move happened: `__unported__/l10n_ch_zip/__openerp__.py → l10n_ch_bank/__openerp__.py`, still, this file is the one in `l10n_ch_bank`.

Comment (yvaucher) [inline l10n_ch_bank/__openerp__.py:24]: Ok @guewen I didn't know it was possible to explicitly give odoo version in module version.

Comment (yvaucher) [inline l10n_ch_bank/__openerp__.py:24]: @open-net-sarl github diff isn't always clever enough to show the right diff. If you have a look at file changed tab, you will see that github doesn't understand where moved the files. What you see here is only the old diff in red.

**Comment by open-net-sarl** [comment]
Bonjour, 

Une petite question bête : pourquoi s'embêter à porter l10n_ch_zip plutôt que d'utiliser base_location_geonames_import , déjà porté et qui supporte les NPA du monde entier !?!? 

Avec mes plus sincères salutations, 

Jean-Marc Vandel 
Senior Project Manager 

Open Net Sàrl 
Rue de l'Industrie 59 
CH-1030 Bussigny 

www.open-net.ch 
+41 79 212.65.94 
jmv@open-net.ch

**Comment by lepistone** [comment]
```
WARNING x openerp.models: @constrains('acc_number', 'bank') parameters must be field names
```

**Comment by guewen** [comment]
> WARNING x openerp.models: @constrains('acc_number', 'bank') parameters must be field names

There is no constrains in these modules. I guess it comes from `l10n_ch_payment_slip`

**Comment by nbessi** [comment]
Hello, 

@guewen  @BT-olivier many thanks for the work.

LGTM, can you please add some tests on l10n_ch_bank before merging.

Regards

Nicolas

**Comment by nbessi** [comment]
Hello, 
Can you please rebase to see if it compatible with https://github.com/OCA/l10n-switzerland/commit/96ab73bba41253ed6b2b375877576b50891c46ca

Regards

Nicolas

**Comment by guewen** [comment]
@nbessi rebase is useless because Travis merge the branch on HEAD of the base before the build. I just restarted the builds.

Regarding the tests, this module is merely importing new data so I don't really see what can be tested in that matter.

**Comment by nbessi** [comment]
@guewen right, I will just restart the travis tests.

**Comment by nbessi** [comment]
:+1:

**Comment by rdeheele** [comment]
:+1:

File: l10n_ch_zip/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Nicolas Bessi, Olivier Jossen, Guewen Baconnier
+#    Copyright Camptocamp SA
+#    Copyright brain-tec AG
```
Comment (yvaucher) [inline l10n_ch_zip/__openerp__.py]: Missing year on copyrights

File: l10n_ch_zip/better_zip.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Nicolas Bessi, Olivier Jossen, Guewen Baconnier
+#    Copyright Camptocamp SA
+#    Copyright brain-tec AG
```
Comment (yvaucher) [inline l10n_ch_zip/better_zip.py]: Same here, missing year

File: l10n_ch_bank/bank.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    bank_group = fields.Char(string='Group', size=2)
+    bank_branchid = fields.Char(string='Branch-ID', size=5)
+    bank_clearing_new = fields.Char(string='BCNr new', size=5)
+    bank_sicnr = fields.Char(string='SIC-Nr', size=6)
+    bank_headquarter = fields.Char(string='Headquarter', size=5)
+    bank_bcart = fields.Char(string='BC-Art', size=1)
+    bank_valid_from = fields.Date(string='Valid from')
+    bank_sic = fields.Char(string='SIC', size=1)
+    bank_eurosic = fields.Char(string='euroSIC', size=1)
+    bank_lang = fields.Char(string='Sprache', size=1)
```
Comment (nbessi) [inline l10n_ch_bank/bank.py]: Language not Sprache

File: l10n_ch_zip/__openerp__.py (hunk 2 above)
Comment (guewen) [inline l10n_ch_zip/__openerp__.py]: Seems to be the case on the whole repository. I added the dates in these modules but one should do another PR to correct the other modules.

**Comment by guewen** [comment]
Fixed the aforementioned points

**Comment by yvaucher** [comment]
Thanks :+1:
